azurerm/resource_arm_app_configuration.go

```diff
@@ -0,0 +1,385 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+	"time"
+
+	appconf "github.com/Azure/azure-sdk-for-go/services/appconfiguration/mgmt/2019-10-01/appconfiguration"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAppConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAppConfigurationCreate,
+		Read:   resourceArmAppConfigurationRead,
+		Update: resourceArmAppConfigurationUpdate,
+		Delete: resourceArmAppConfigurationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		SchemaVersion: 2,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAppConfigurationName,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"sku": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "free",
+				ValidateFunc: validation.StringInSlice([]string{
+					"free",
+					"standard",
+				}, false),
+			},
+
+			"endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"primary_read_key": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
```

Comment:
This can be removed as `MaxItems` is a validation check for what a user submits rather than a check that the api only returns 1 of this. This goes for `MaxItems` on the blocks down below as well

Reply:
Just out of curiosity, I got this from [resource_arm_kubernetes_cluster.go](https://github.com/terraform-providers/terraform-provider-azurerm/blob/fc2de5e6850eb5a7ed68a030c67e11fe83fb40f0/azurerm/resource_arm_kubernetes_cluster.go#L525). Is there some reason why it's necessary there?

Reply:
Not that I'm aware! It might just be a mistake. That validation is only checked against what a user submits not what the api returns.